contracts/0.8.25/vaults/OperatorGrid.sol

```diff
@@ -0,0 +1,448 @@
+    /// @param _locator Lido Locator contract
+    constructor(ILidoLocator _locator) {
+        LIDO_LOCATOR = _locator;
+
+        _disableInitializers();
+    }
+
+    function initialize(address _admin) external initializer {
+        if (_admin == address(0)) revert ZeroArgument("_admin");
+
+        __AccessControlEnumerable_init();
+        _grantRole(DEFAULT_ADMIN_ROLE, _admin);
+
+        ERC7201Storage storage $ = _getStorage();
+
+        $.tiers.push(Tier(address(0), 0, 0, 0, 0, 0));
+    }
+
+    /// @notice Registers a new group
+    /// @param _nodeOperator address of the node operator
+    /// @param _shareLimit Maximum share limit for the group
+    function registerGroup(address _nodeOperator, uint256 _shareLimit) external onlyRole(REGISTRY_ROLE) {
+        if (_nodeOperator == address(0)) revert ZeroArgument("_nodeOperator");
+
+        ERC7201Storage storage $ = _getStorage();
+        if ($.groups[_nodeOperator].operator != address(0)) revert GroupExists();
+
+        $.groups[_nodeOperator] = Group({
+            operator: _nodeOperator,
+            shareLimit: uint96(_shareLimit),
+            mintedShares: 0,
+            tiersId: new uint256[](0)
+        });
+        $.nodeOperators.push(_nodeOperator);
+
+        emit GroupAdded(_nodeOperator, uint96(_shareLimit));
+    }
+
+    /// @notice Updates the share limit of a group
+    /// @param _nodeOperator address of the node operator
+    /// @param _shareLimit New share limit value
+    function updateGroupShareLimit(address _nodeOperator, uint256 _shareLimit) external onlyRole(REGISTRY_ROLE) {
+        if (_nodeOperator == address(0)) revert ZeroArgument("_nodeOperator");
+
+        ERC7201Storage storage $ = _getStorage();
+        Group storage group_ = $.groups[_nodeOperator];
+        if (group_.operator == address(0)) revert GroupNotExists();
+
+        group_.shareLimit = uint96(_shareLimit);
+
+        emit GroupShareLimitUpdated(_nodeOperator, uint96(_shareLimit));
+    }
+
+    /// @notice Returns a group by ID
+    /// @param _nodeOperator address of the node operator
+    /// @return Group
+    function group(address _nodeOperator) external view returns (Group memory) {
+        return _getStorage().groups[_nodeOperator];
+    }
+
+    /// @notice Returns a node operator address by index
+    /// @param _index index of the node operator
+    /// @return Node operator address
+    function nodeOperatorAddress(uint256 _index) external view returns (address) {
+        return _getStorage().nodeOperators[_index];
+    }
+
+    /// @notice Returns a node operator count
+    /// @return Node operator count
+    function nodeOperatorCount() external view returns (uint256) {
+        return _getStorage().nodeOperators.length;
+    }
+
+    /// @notice Registers a new tier
+    /// @param _nodeOperator address of the node operator
+    /// @param _tiers array of tiers to register
+    function registerTiers(
```

Comment:
shall we check that cumulative share limit should be below the group's? :thinking:

Reply:
probably discuss with Max

Reply:
probably not